core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateConfigBuildStep.java

```diff
@@ -112,8 +113,23 @@ void build(SslContextConfigurationTemplate sslContextConfigurationTemplate,
                 }
             }
         }
-
         nativeImage.produce(new SubstrateSystemPropertyBuildItem("quarkus.ssl.native", sslNativeEnabled.toString()));
+
+        if (!jniBuildItems.isEmpty()) {
+            boolean enableJni = false;
+            for (JniBuildItem jniBuildItem : jniBuildItems) {
+                enableJni = true;
```

Comment:
Hmmm, do we need `enableJni`? I mean it will always be true if `jniBuildItems` is not empty.